bin/p3.go

```diff
@@ -0,0 +1,55 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"log"
+	"os"
+
+	"github.com/Clever/pathio"
+	"gopkg.in/alecthomas/kingpin.v2"
+)
+
+var (
+	action    = kingpin.Arg("action", "S3 action: `upload` or `download`").Required().String()
+	s3Path    = kingpin.Arg("s3_path", "S3 path to upload/download").Required().String()
+	localPath = kingpin.Arg("local_path", "local file to upload or write to").Required().String()
+)
+
+func main() {
+	kingpin.Parse()
```

Comment:
kingpin handles providing a usage and doing all the required vs optional enforcement and type enforcement. Seems nicely convenient. The saving is small in this project, so I'm happy to remove it and use `flag` instead if you prefer.